### PR TITLE
feat: gen secrets from kubernetes pki dir

### PR DESCRIPTION
**Pull request by utkuozdemir**
This PR allows the ability to generate `secrets.yaml` (`talosctl gen secrets`) using a Kubernetes PKI directory path (e.g. `/etc/kubernetes/pki`) as input. Also introduces the flag `--kubernetes-bootstrap-token` to be able to set a static Kubernetes bootstrap token to the generated `secrets.yaml` file instead of a randomly-generated one. Closes siderolabs/talos#5894.

Signed-off-by: Utku Ozdemir <utku.ozdemir@siderolabs.com>

### CODE REVIEW

File: internal/integration/cli/pki.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	//go:embed "testdata/pki/sa.key"
+	pkiSAKey []byte
+	//go:embed "testdata/pki/etcd/ca.crt"
+	pkiEtcdCACrt []byte
+	//go:embed "testdata/pki/etcd/ca.key"
+	pkiEtcdCAKey []byte
+)
+
+func createTempPKIDir() (string, error) {
+	dir, err := os.MkdirTemp("", "talos-gen-test-")
```
Comment (smira) [inline internal/integration/cli/pki.go:31]: we can do `suite.T().TempDir()` which has automatic cleanup

File: pkg/machinery/config/types/v1alpha1/generate/generate.go [hunk 2]
```diff
@@ -208,41 +208,26 @@ func NewSecretsBundle(clock Clock, opts ...GenOption) (*SecretsBundle, error) {
...
 	}
 
+	return NewSecretsBundleWithOptions(clock, options.VersionContract, "", "")
+}
+
+// NewSecretsBundleWithOptions creates secrets bundle,
+// generating all secrets or reading from the input options if provided.
+//
+//nolint:gocyclo
+func NewSecretsBundleWithOptions(clock Clock, versionContract *config.VersionContract, pkiDir, bootstrapToken string) (*SecretsBundle, error) {
```
Comment (smira) [inline pkg/machinery/config/types/v1alpha1/generate/generate.go:218]: from the API perspective it feels like it should be a different method.

we have `NewSecretsBundle` which gives us all fresh unique PKI
we can have `NewSecretsBundleFromKubernetesPKI` which imports whatever it can from Kubernetes PKI and fills in the missing stuff. 

The implementation might be that we have a method which fills in missing secrets (generates them), and the `NewSecretsBundle` just calls it, while `NewSecretsBundleFromKubernetesPKI` first fills in what it can, and that calls that common shared method?

Right now it feels a bit of a hard to follow code, and this code is very critical to be reviewable

File: internal/integration/cli/gen.go [hunk 3]
```diff
@@ -209,19 +208,62 @@ func (suite *GenSuite) TestSecrets() {
...
+// TestSecretsWithPKIDirAndToken ...
+func (suite *GenSuite) TestSecretsWithPKIDirAndToken() {
+	path := "/tmp/secrets-with-pki-dir-and-token.yaml"
+
+	dir, err := createTempPKIDir()
+	suite.Assert().NoError(err)
+
+	defer os.RemoveAll(dir) //nolint:errcheck
+
+	suite.RunCLI([]string{
```
Comment (smira) [inline internal/integration/cli/gen.go:233]: :+1: on having these tests

File: internal/integration/cli/pki.go (hunk 1 above)
Comment (utkuozdemir) [reply internal/integration/cli/pki.go]: done

File: pkg/machinery/config/types/v1alpha1/generate/generate.go (hunk 2 above)
Comment (utkuozdemir) [reply pkg/machinery/config/types/v1alpha1/generate/generate.go]: I did it the way you suggested, you can have another look.

**Review by frezbo — APPROVED** [review]
Also needs something in the release notes and also a knowledgeable docs

**Review by frezbo — APPROVED** [review]

**Comment by utkuozdemir** [comment]
Its documentation will be covered in a separate PR.

**Review by smira — APPROVED** [review]
:+1:
